examples/chef/common/clusters/media-input/MediaInputManager.cpp

```diff
@@ -0,0 +1,106 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/config.h>
+#ifdef MATTER_DM_PLUGIN_MEDIA_INPUT_SERVER
+#include "MediaInputManager.h"
+
+using namespace std;
+using namespace chip;
+using namespace chip::app::Clusters::MediaInput;
+
+MediaInputManager::MediaInputManager()
+{
+    mCurrentInput = 1;
+
+    for (int i = 1; i < mTotalInput; ++i)
+    {
+        InputInfoType inputInfo;
+        inputInfo.description = chip::CharSpan::fromCharString("High-Definition Multimedia Interface");
+        inputInfo.name        = chip::CharSpan::fromCharString("HDMI");
+        inputInfo.inputType   = chip::app::Clusters::MediaInput::InputTypeEnum::kHdmi;
+        inputInfo.index       = static_cast<uint8_t>(i);
+        mInputs.push_back(inputInfo);
+    }
+}
+
+CHIP_ERROR MediaInputManager::HandleGetInputList(chip::app::AttributeValueEncoder & aEncoder)
+{
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        for (auto const & inputInfo : this->mInputs)
+        {
+            ReturnErrorOnFailure(encoder.Encode(inputInfo));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+uint8_t MediaInputManager::HandleGetCurrentInput()
+{
+    return mCurrentInput;
+}
+
+bool MediaInputManager::HandleSelectInput(const uint8_t index)
+{
+    for (InputInfoType & input : mInputs)
+    {
+        if (input.index == index)
+        {
+            mCurrentInput = index;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool MediaInputManager::HandleShowInputStatus()
+{
+    ChipLogProgress(Zcl, " MediaInputManager::HandleShowInputStatus()");
+    for (auto const & inputInfo : this->mInputs)
+    {
+        string name(inputInfo.name.data(), inputInfo.name.size());
+        string desc(inputInfo.description.data(), inputInfo.description.size());
+        ChipLogProgress(Zcl, " [%d] type=%d selected=%d name=%s desc=%s", inputInfo.index,
+                        static_cast<uint16_t>(inputInfo.inputType), (mCurrentInput == inputInfo.index ? 1 : 0), name.c_str(),
+                        desc.c_str());
+    }
+    return true;
+}
+
+bool MediaInputManager::HandleHideInputStatus()
+{
+    ChipLogProgress(Zcl, " MediaInputManager::HandleHideInputStatus()");
+    return true;
+}
+
+bool MediaInputManager::HandleRenameInput(const uint8_t index, const chip::CharSpan & name)
+{
+    for (InputInfoType & input : mInputs)
+    {
+        if (input.index == index)
+        {
+            const size_t len = std::min(mNameLenMax, name.size());
```

Comment:
same comment here: if you inline without a helper, you need to test index being valid.

Reply:
Same reply as above: It seems to be safe since I'll check `input.index == index`.

Reply:
mInputs is a vector of varying size. While the code is correct now, it is more brittle: it does not allow mInputs to change too much, so it is not obviously always correct.

Reply:
Fixed according to the proposal. Thanks